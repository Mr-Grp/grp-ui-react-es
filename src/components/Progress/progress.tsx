import React, { FC } from 'react';

export type ThemeProps = 'primary' | 'secondary' | 'success' | 'info' | 'warning' | 'danger' | 'light' | 'dark'

export interface ProgressProps {
  percent: number;
  strokeHeight?: number;
  showText?: boolean;
  styles?: React.CSSProperties;
  theme?: ThemeProps;
}

const Progress: FC<ProgressProps> = (props) => {
  const {
    percent,
    strokeHeight = 15,
    showText = true,
    styles,
    theme = 'primary',
  } = props;
  return (
    <div className="progress-bar" style={styles} data-test="progress">
      <div className="progress-bar-outer" style={{ height: `${strokeHeight}px` }} data-test="progress-outer">
        <div
          className={`progress-bar-inner color-${theme}`}
          style={{ width: `${percent}%` }}
          data-test="progress-inner"
        >
          {showText && <span className="inner-text" data-test="progress-inner-text">{`${percent}%`}</span>}
        </div>
      </div>
    </div>
  );
};

export default Progress;
