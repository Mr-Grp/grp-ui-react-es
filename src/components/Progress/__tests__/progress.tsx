import { shallow } from 'enzyme';
import Progress from '..';
import { findTestWrapper } from '../../../utils/testUtil';

describe('Progress 测试', () => {
  test('默认进度条应该正常展示', () => {
    const wrapper = shallow(<Progress percent={10} />);
    const progressElem = findTestWrapper(wrapper, 'progress');
    const progressOuterElem = findTestWrapper(wrapper, 'progress-outer');
    const progressInnerElem = findTestWrapper(wrapper, 'progress-inner');
    const progressInnerTextElem = findTestWrapper(wrapper, 'progress-inner-text');
    expect(progressElem).toExist();
    expect(progressOuterElem).toExist();
    expect(progressInnerElem).toExist();
    expect(progressInnerTextElem).toExist();
    expect(progressInnerElem.prop('style')).toMatchObject({ width: '10%' });
    expect(progressInnerElem.prop('className')).toContain('color-primary');
    expect(progressOuterElem.prop('style')).toMatchObject({ height: '15px' });
    expect(progressInnerTextElem.text()).toContain('10%');
  });

  test('进度条可以传入其他风格', () => {
    const wrapper = shallow(
      <Progress percent={10} strokeHeight={20} theme="danger" />,
    );
    const progressOuterElem = findTestWrapper(wrapper, 'progress-outer');
    const progressInnerElem = findTestWrapper(wrapper, 'progress-inner');
    expect(progressOuterElem.prop('style')).toMatchObject({ height: '20px' });
    expect(progressInnerElem.prop('className')).toContain('color-danger');
  });

  test('可以不展示百分比文字', () => {
    const wrapper = shallow(
      <Progress percent={10} showText={false} />,
    );
    const progressInnerTextElem = findTestWrapper(wrapper, 'progress-inner-text');
    expect(progressInnerTextElem).not.toExist();
  });

  test('测试完成，保存快照', () => {
    const wrapper = shallow(
      <Progress percent={10} showText={false} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
