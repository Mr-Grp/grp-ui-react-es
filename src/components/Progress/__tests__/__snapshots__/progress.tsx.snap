// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress 测试 测试完成，保存快照 1`] = `
<div
  className="progress-bar"
  data-test="progress"
>
  <div
    className="progress-bar-outer"
    data-test="progress-outer"
    style={
      Object {
        "height": "15px",
      }
    }
  >
    <div
      className="progress-bar-inner color-primary"
      data-test="progress-inner"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;
